src/Utils/FileUtils.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2021 The Foedag team
+
+GPL License
+
+Copyright (c) 2021 The Open-Source FPGA Foundation
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "Utils/FileUtils.h"
+
+#include <errno.h>
+#include <limits.h> /* PATH_MAX */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+
+#include <algorithm>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <regex>
+#include <sstream>
+#include <string>
+
+namespace FOEDAG {
+
+namespace fs = std::filesystem;
+
+bool FileUtils::fileExists(const fs::path& name) {
+  std::error_code ec;
+  return fs::exists(name, ec);
+}
+
+uint64_t FileUtils::fileSize(const fs::path& name) {
+  std::error_code ec;
+  return fs::file_size(name, ec);
+}
+
+bool FileUtils::fileIsDirectory(const fs::path& name) {
+  return fs::is_directory(name);
+}
+
+bool FileUtils::fileIsRegular(const fs::path& name) {
+  return fs::is_regular_file(name);
+}
+
+bool FileUtils::mkDirs(const fs::path& path) {
+  // CAUTION: There is a known bug in VC compiler where a trailing
+  // slash in the path will cause a false return from a call to
+  // fs::create_directories.
+  std::error_code err;
+  fs::create_directories(path, err);
+  return fs::is_directory(path);
+}
+
+bool FileUtils::rmDirRecursively(const fs::path& path) {
+  static constexpr uintmax_t kErrorCondition = static_cast<std::uintmax_t>(-1);
+  std::error_code err;
+  return fs::remove_all(path, err) != kErrorCondition;
+}
+
+fs::path FileUtils::getFullPath(const fs::path& path) {
+  std::error_code ec;
+  fs::path fullPath = fs::canonical(path, ec);
+  return ec ? path : fullPath;
+}
+
+bool FileUtils::getFullPath(const fs::path& path, fs::path* result) {
+  std::error_code ec;
+  fs::path fullPath = fs::canonical(path, ec);
+  bool found = (!ec && fileIsRegular(fullPath));
+  if (result != nullptr) {
+    *result = found ? fullPath : path;
+  }
+  return found;
+}
+
+std::string FileUtils::getFileContent(const fs::path& filename) {
```

Comment:
I suggest to change return value as **std::pair<std::string, bool>** in order to understand if there is any error.